### DISCUSSION TITLE
VCPKG_DEFAULT_BINARY_CACHE VS VCPKG_BINARY_SOURCES

### DISCUSSION THREAD

**Discussion by AbdulAbCellera** [microsoft/vcpkg#29226] [Q&A] [answered]
A year and a half of using vcpkg, and I am still confused by the documentation of those two env/variables. Could any one explain them, please?

**Comment by JackBoosY** [comment]
VCPKG_DEFAULT_BINARY_CACHE : to determine which binary cache to restore directly (won't build).
VCPKG_BINARY_SOURCES : to determine which source package to build the binaries.

**Comment by hamaney** [comment, marked as answer]
Thanks for replying. Could you elaborate?

**Reply by JackBoosY** [reply]
```
run install command
-> check the binary cache which determine by VCPKG_DEFAULT_BINARY_CACHE
    -> have binary cache
        -> install
        -> end
    -> don't have binary cache
        -> check the binary source which determine by VCPKG_BINARY_SOURCES
            -> don't have binary source
                -> download source package
            -> have binary source
                -> directly extract the binary source to build_trees folder.
        -> build port
        -> install
        -> end
```

**Reply by AbdulAbCellera** [reply]
Thank you a lot. If I knew it would only take me to ask, I would have done it a year ago. I feel your answer should be added to the docs.
